client/organization_factory.go

```diff
@@ -0,0 +1,218 @@
+}
+
+func (f *OrganizationFactory) GetVirtualMachineScaleSetsClient(ctx context.Context, objectMeta v1.ObjectMeta) (*compute.VirtualMachineScaleSetsClient, error) {
+	credentialSecret, err := f.getCredentialSecret(ctx, objectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return f.factory.GetVirtualMachineScaleSetsClient(credentialSecret.Namespace, credentialSecret.Name)
+}
+
+func (f *OrganizationFactory) GetVirtualMachineScaleSetVMsClient(ctx context.Context, objectMeta v1.ObjectMeta) (*compute.VirtualMachineScaleSetVMsClient, error) {
+	credentialSecret, err := f.getCredentialSecret(ctx, objectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return f.factory.GetVirtualMachineScaleSetVMsClient(credentialSecret.Namespace, credentialSecret.Name)
+}
+
+func (f *OrganizationFactory) GetVirtualNetworksClient(ctx context.Context, objectMeta v1.ObjectMeta) (*network.VirtualNetworksClient, error) {
+	credentialSecret, err := f.getCredentialSecret(ctx, objectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return f.factory.GetVirtualNetworksClient(credentialSecret.Namespace, credentialSecret.Name)
+}
+
+func (f *OrganizationFactory) GetSnapshotsClient(ctx context.Context, objectMeta v1.ObjectMeta) (*compute.SnapshotsClient, error) {
+	credentialSecret, err := f.getCredentialSecret(ctx, objectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return f.factory.GetSnapshotsClient(credentialSecret.Namespace, credentialSecret.Name)
+}
+
+func (f *OrganizationFactory) GetStorageAccountsClient(ctx context.Context, objectMeta v1.ObjectMeta) (*storage.AccountsClient, error) {
+	credentialSecret, err := f.getCredentialSecret(ctx, objectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return f.factory.GetStorageAccountsClient(credentialSecret.Namespace, credentialSecret.Name)
+}
+
+func (f *OrganizationFactory) GetSubnetsClient(ctx context.Context, objectMeta v1.ObjectMeta) (*network.SubnetsClient, error) {
+	credentialSecret, err := f.getCredentialSecret(ctx, objectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return f.factory.GetSubnetsClient(credentialSecret.Namespace, credentialSecret.Name)
+}
+
+func (f *OrganizationFactory) GetNatGatewaysClient(ctx context.Context, objectMeta v1.ObjectMeta) (*network.NatGatewaysClient, error) {
+	credentialSecret, err := f.getCredentialSecret(ctx, objectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return f.factory.GetNatGatewaysClient(credentialSecret.Namespace, credentialSecret.Name)
+}
+
+func (f *OrganizationFactory) getCredentialSecret(ctx context.Context, objectMeta v1.ObjectMeta) (*v1alpha1.CredentialSecret, error) {
+	f.logger.LogCtx(ctx, "level", "debug", "message", "finding credential secret")
+
+	secretList := &corev1.SecretList{}
+	{
+		err := f.ctrlClient.List(
+			ctx,
+			secretList,
+			client.InNamespace(objectMeta.Namespace),
```

Comment:
This is the important part. Instead of using the default namespace, here we are using the source CR namespace.